### ISSUE TITLE
Runtime thread names

### ISSUE THREAD

**Issue by macaba** [dotnet/runtime#34276]
Running an application on Ubuntu 18.04.4 LTS under .NET 5 Preview 1 using Server GC, then running `htop` and enabling custom thread names gives me this view:

![putty_2020-03-29_19-22-21](https://user-images.githubusercontent.com/1031306/77912169-f84b6c80-7289-11ea-89b3-e8bdc61ddaba.png)

Most of the threads make sense to me, but there seems to be 7 unnamed threads (`GeoVR.RangingSe`) some of which I suspect are GC threads.

It would be beneficial if the runtime named these threads.

I would be willing to attempt a pull request but I have no runtime development experience so any hints to where I should be looking in the codebase would be most appreciated.

**Comment by jkotas** [comment]
Related https://github.com/dotnet/coreclr/pull/27182

**Comment by macaba** [comment]
The plot thickens; on a different service that I just moved to .NET 5, there are threads named `.NET Background` and `.NET Server GC` and still 7 unnamed threads, so that would seem to eliminate the server GC as having unnamed threads.

**Comment by stephentoub** [comment]
> so any hints to where I should be looking in the codebase would be most appreciated.

Can you attach a debugger/grab a dump and look at the call stacks of those threads?

**Comment by macaba** [comment]
I was able to attach a debugger over SSH.

First observation: The 'threads' view in Visual Studio shows a lot of unnamed threads that are actually named in `htop` (like `.NET ThreadPool`).

So I moved over to `dotnet-dump` and was able to `analyze`. I'm new to `dotnet-dump` but I was able to ascertain that all the unnamed threads are unmanaged threads. Is there a sequence of commands for showing more information about unmanaged threads?
